src/rust_backend.rs

```diff
@@ -318,8 +323,9 @@ impl Bn {
 
 #[test]
 fn safe_prime() {
-    let n = Bn::safe_prime(1024);
-    assert_eq!(n.0.bits(), 1024);
+    let n = Bn::safe_prime(128);
+    // TODO: Rust backend samples a prime from (0, 2^(n+1)) and so doesn't guarantee the bit length
+    assert!(n.bit_length() <= 128 + 1);
```

Comment:
why `128+1`?

Reply:
The rust backend (implemented in another crate) doesn't generate a safe prime of the bit length specified. It's optimization can generate a value one bit larger.